### PR TITLE
Hald done

### PR DESCRIPTION
**Pull request by JackCho**

### CODE REVIEW

File: amigo-lib/src/main/java/me/ele/amigo/release/ApkReleaseActivity.java [hunk 1]
```diff
@@ -79,29 +85,36 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            service.submit(new Runnable() {
+                @Override
+                public void run() {
+                    long startTime = System.currentTimeMillis();
+                    new DexClassLoader(dex.getAbsolutePath(), optimizedDir.getAbsolutePath(), null, DexUtils.getPathClassLoader());
+                    Log.e(TAG, String.format("dex %s consume %d ms", dex.getAbsolutePath(), System.currentTimeMillis() - startTime));
+                    int num = counter.incrementAndGet();
+                    if (num == validDexes.size()) {
+                        handler.sendEmptyMessage(WHAT_DEX_OPT_DONE);
+                    }
```
Comment (FtheGFW) [inline amigo-lib/src/main/java/me/ele/amigo/release/ApkReleaseActivity.java]: 这儿可以使用CountDownLatch来简化一下

File: amigo-lib/src/main/java/me/ele/amigo/release/ApkReleaseActivity.java [hunk 2]
```diff
@@ -79,29 +85,36 @@ protected void onCreate(Bundle savedInstanceState) {
             @Override
             public void run() {
                 super.run();
-
                 DexReleaser.releaseDexes(demoAPk.getAbsolutePath(), dexDir.getAbsolutePath());
                 NativeLibraryHelperCompat.copyNativeBinaries(demoAPk, nativeLibraryDir);
                 dexOptimization();
```
Comment (FtheGFW) [inline amigo-lib/src/main/java/me/ele/amigo/release/ApkReleaseActivity.java]: 可以用上面声明的ExecutorService来改写这儿的new thread的

File: amigo-lib/src/main/java/me/ele/amigo/release/ApkReleaseActivity.java (hunk 1 above)
Comment (FtheGFW) [inline amigo-lib/src/main/java/me/ele/amigo/release/ApkReleaseActivity.java]: ExecutorService这儿都可以shutdown了吧

Comment (peacepassion) [inline amigo-lib/src/main/java/me/ele/amigo/release/ApkReleaseActivity.java]: 上面会把这个进程关掉

Comment (FtheGFW) [inline amigo-lib/src/main/java/me/ele/amigo/release/ApkReleaseActivity.java]: Initiates an orderly shutdown in which previously submitted
 tasks are executed, but no new tasks will be accepted.

**Comment by JackCho** [comment]
@FtheGFW review again

**Comment by FtheGFW** [comment]
👍

**Pull request merged**